build-support/bin/mypy.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import subprocess
+from typing import List
+
+from common import die
+
+
+def main():
+  targets = create_parser().parse_args().targets
+  run_mypy(targets)
+
+
+def create_parser() -> argparse.ArgumentParser:
+  parser = argparse.ArgumentParser(description="Run MyPy with our config file.")
+  parser.add_argument(
+    "targets",
+    default=["build-support::", "src/python::", "tests/python::", "contrib::"],
+    nargs="*",
+    help="Pants targets, e.g. `tests::`.",
+  )
+  return parser
+
+
+def run_mypy(targets: List[str]) -> None:
+  command = [
+    "./pants",
+    "mypy",
+    "--mypy-mypy-version=0.701",
+    "--config-file=build-support/mypy/mypy.ini",
+  ]
+  try:
+    subprocess.run(command + targets, check=True)
```

Comment:
Not sure which one I prefer, but iirc `subprocess.check_call(command + targets)` does the same thing

Reply:
In Python 3.5+ land, you almost always want to use `subprocess.run()`. It can handle almost every instance you can imagine, beyond async calls. I recommend skimming https://codecalamity.com/run-subprocess-run/.

You're right that `subprocess.check_call()` would work, but it's better for us to converge on always using `subprocess.run()` where possible for consistency in our codebase.

Reply:
Thanks for the reference article :) I'm sure which one I prefer now.